src/contracts/mod.rs

```diff
@@ -204,6 +218,26 @@ impl IdentityManager {
         Ok(proof_data)
     }
 
+    #[instrument(level = "debug", skip(prover, identity_commitments))]
+    pub async fn prepare_deletion_proof(
```

Comment:
The signature for getting the proof will be different. Instead of commitments we'll use leaf indexes.

Refer to https://github.com/worldcoin/semaphore-mtb/blob/87c50e4eb609e6fa80e70ff16e8d00b93deeb384/prover/deletion_proving_system.go#L16

Reply:
Correction: we'll need both the commitments and indexes